src/Microsoft.AspNetCore.Authentication.AzureAd/AzureAdB2CInitializeOptions.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Authentication.AzureAd
+{
+    internal class AzureAdB2CInitializeOptions : IInitializeOptions<AzureAdB2COptions>
+    {
+
+        public AzureAdB2CInitializeOptions() { }
+
+        public void Initialize(string name, AzureAdB2COptions options)
+        {
+            if (string.IsNullOrEmpty(options.Authority))
+            {
+                // Bind to only to any of the AzureAdB2C policy schemes
+                if (name == AzureDefaults.AzureAdB2CEditProfileAuthenticationScheme)
```

Comment:
Instead of special casing these three policies I'd prefer to have a generic Policies dictionary where the key is the authentication scheme name for that policy and the value is the policy ID you setup with Azure AD B2C.